docs/docs/explanations/architecture.mdx

```diff
@@ -0,0 +1,112 @@
+---
+id: infrastructure-architecture
+title: Nebari architecture and conceptual guide
+---
+
+# Nebari architecture and conceptual guide
+
+Nebari relies on a few core concepts to make it work.
+This guide will help you understand how Nebari works and how to use it.
+
+Some of the most important concepts that Nebari relies on are Terraform and Kubernetes.
+
+![Nebari core concepts](/img/explanations/core-concepts.png)
+
+## Terraform
+
+Terraform is an excellent option for deploying and managing cloud
+infrastructure. It allows you to easily configure any number of
+cloud servers as well as all the required connections and features
+to ensure security, redundancy and scalability
+
+Nebari uses Terraform for managing its infrastructure in configuration
+files that are easily accessible and extendable in our code base, these
+files acts as the single source of truth for all the components required
+in the Infrastructure, such as Load Balancers, Virtual Machines, Gateways,
+etc. Such methodology is usualy referred to as Infrastructure as code (IaC).
+
+For more information about Terraform, IaC and the features it provides, refer to
+[Infrastructure as Code with Terraform](https://learn.hashicorp.com/tutorials/terraform/infrastructure-as-code).
+
+![Terraform](/img/explanations/terraform.png)
+
+## Kubernetes
+
+Kubernetes is a powerful open-source system, initially developed by Google,
+for managing containerized applications in a clustered environment. It aims
+to provide better ways of managing related, distributed components and services
+across varied infrastructure.
+
+Kubernetes provides fine-grained control in applications and container lifecycles,
+especially in dynamic environments. Tasks such as pod tracking, availability, and
```

Comment:
I wouldn't write "pod tracking" here. We're explaining Kubernetes here and assuming people would know what's a pod is probably not a good idea and "pod tracking" isn't really a popular term anyway.